pkgs/development/libraries/openmpi/default.nix

```diff
@@ -51,19 +56,29 @@ stdenv.mkDerivation rec {
     ++ lib.optionals cudaSupport [ cudaPackages.cuda_nvcc ]
     ++ lib.optionals fortranSupport [ gfortran ];
 
-  configureFlags = lib.optional (!cudaSupport) "--disable-mca-dso"
+  configureFlags =
+    [
+      "--with-ucc=${lib.getDev ucc}"
+      "--with-ucx=${lib.getDev ucx}"
+      "--with-ucx-libdir=${lib.getLib ucx}/lib"
```

Comment:
I doubted because of:

```
❯ nix log nixpkgs#openmpi > old.log
❯ nix log .#openmpi > new.log
❯ diff old.log new.log
...
> checking looking for OFI libfabric in... (/nix/store/4crym3mdnp24333x11yv3wzlkak5lnci-libfabric-1.20.0-dev)
1263,1265c1284,1285
< checking --with-ucx value... simple ok (unspecified value)
< checking --with-ucx-libdir value... simple ok (unspecified value)
< checking for ucx... no
---
> checking --with-ucx value... sanity check ok (/nix/store/q534s1m6rhj6p6774bfaaq9jw126jd5i-ucx-1.15.0-dev)
> checking --with-ucx-libdir value... sanity check ok (/nix/store/sik8ws5jfzkxdlbw9rzf353fwpzcp1br-ucx-1.15.0/lib)
1269d1288
...
```

Reply:
Also the comment below, `# TODO: add UCX support, which is recommended to use with cuda for the most robust OpenMPI build`

Reply:
Made them explicit `bool` flags, paths discovered by autotools from `buildInputs`